client/state/notices/types.ts

```diff
@@ -0,0 +1,70 @@
+/**
+ * External dependencies
+ */
+import type { ReactNode } from 'react';
+
+export type NoticeId = string;
+
+export type NoticeStatusError = 'is-error';
+export type NoticeStatusInfo = 'is-info';
+export type NoticeStatusPlain = 'is-plain';
+export type NoticeStatusSuccess = 'is-success';
+export type NoticeStatusWarning = 'is-warning';
+export type NoticeStatus =
+	| NoticeStatusError
+	| NoticeStatusInfo
+	| NoticeStatusPlain
+	| NoticeStatusSuccess
+	| NoticeStatusWarning;
+
+export type NoticeText = string | ReactNode;
+
+// Notice options that are always the same
+export interface BaseNoticeOptions {
+	button?: string;
+	displayOnNextPage?: boolean;
+	duration?: null | number;
+	href?: string;
+	isPersistent?: boolean;
+	onClick?: () => void;
+	showDismiss?: boolean;
+}
+
+// Notice options as they're expected by the action creator
+export interface NoticeOptions extends BaseNoticeOptions {
+	id?: NoticeId;
+}
+
+// Notice options as they're returned by the action creator
+export interface NoticeActionOptions extends BaseNoticeOptions {
+	noticeId: NoticeId;
+	status: NoticeStatus;
+	text: NoticeText;
+}
```

Comment:
Could this interface be turned into a generic one where the status is specific? Could that render any benefit or would it just complicate things without benefiting it?

Reply:
I don't see any benefit coming from making these generic by status - while we benefit from the fact that we type all those options, the rest of the properties are the same, and I believe that by providing a more general type for all notices (vs separating them by status) we allow for better flexibility. Also, TBH, I don't see much value in such generic typing from a consumer point of view. We can always update that if we eventually end up needing granularity, though.